docs/upgrades.md

```diff
@@ -0,0 +1,196 @@
+# Upgrades
+
+This page contains some guidance on upgrading an existing deployment of Sematic
+in various circumstances.
+
+## General guidance
+
+Sematic has 3 components, depending on your usage:
```

Comment:
`pip package` and `bazel integration` aren't "components" per se. How about "version configurations", "versioned aspects", or "versioned dependencies"?